src/Microsoft.Restier.WebApi/Query/ODataCountOption.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Restier.WebApi.Query
+{
+    internal class ODataCountOption
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether the total
+        /// number of items should be retrieved when the
+        /// result has been filtered using paging operators.
+        /// </summary>
+        /// <remarks>
+        /// Setting this to <c>true</c> may have a performance impact as
+        /// the data provider may need to execute two independent queries.
+        /// </remarks>
+        public bool IncludeTotalCount { get; set; }
+
+        public Action<long> SetTotalCount { get; set; }
+    }
+}
```

Comment:
Yeah, #326 is just a prototype, and not synced to head. Sorry for wasting your review time.
You mean HttpRequestMessage.Properties?, Is there anyway Restier query pipeline can get the HttpRequestMessage?